autopush/db.py

```diff
@@ -57,13 +57,18 @@ def get_storage_table():
 
 
 class Storage(object):
-    def __init__(self, table):
+    def __init__(self, table, metrics):
         self.table = table
+        self.metrics = metrics
 
     def fetch_notifications(self, uaid):
-        notifs = self.table.query_2(consistent=True, uaid__eq=uaid,
-                                    chid__gt=" ")
-        return list(notifs)
+        try:
+            notifs = self.table.query_2(consistent=True, uaid__eq=uaid,
+                                        chid__gt=" ")
+            return list(notifs)
+        except ProvisionedThroughputExceededException:
```

Comment:
I'm glad we're noting exceptions like ProvisionedThroughput, but there are other exceptions we might want to be aware of (ResourceInUse, ItemCollectionSizeLimitExceeded, etc.) Do we know if there's enough info in these to analyze the cause if they're handled by the default catch mechanism?